desk/lib/channel-utils.hoon

```diff
@@ -429,13 +429,35 @@
     id       id.seal
     reacts   (reacts-1 reacts.seal)
     replies  (replies-1 replies.seal)
-    reply-meta  reply-meta.seal
+    reply-meta  (reply-meta-1 reply-meta.seal)
   ==
 ::
+++  author-1
+  |=  =author:c
+  ^-  ship
+  ?@  author  author
+  ship.author
+::XX fill in
+++  memo-1
+  |=  memo:c
+  ^-  memo:v7:old:c
+  *memo:v7:old:c
+::
+++  essay-1
+  |=  =essay:c
+  ^-  essay:v7:old:c
+  *essay:v7:old:c
+::
+++  rev-essay-1
+  |=  essay=(rev:c essay:c)
+  ^-  (rev:c essay:v7:old:c)
+  [rev.essay *essay:v7:old:c]
+::
```

Comment:
seems like these need to be filled in

Reply:
For now I have implemented down-conversion of existing post kinds, that in the new scheme correspond to kind paths `/chat`, `/chat/notice`, `/diary` and `/heap`. For posts that are not of this kind we just crash, which means such updates won't make it to an old client.

Reply:
Understood. As long as we maintain some kind of basic backcompat until we release the new desktop client, I'm OK with some signal degradation between clients. 

When you say crash—will the old client stop receiving updates to the post? E.g. edits, reactions, replies, etc

Reply:
> We can always hide channels that are not of those kind on the front-end of the old client.
To be sure we are on the same page here: this is about the post kind. A channel in principle can contain a mixture of different post kinds. 

There is also another https://github.com/tloncorp/tlon-apps/pull/4239 that introduces the channel meta field which carries arbitrary channel metadata.